File: 프로그래머스/2/12981. 영어 끝말잇기/영어 끝말잇기.py
def solution(n, words):
    answer = [0,0]
    dic = {}
    cnt = [0] * n
    last = ''
    for idx, word in enumerate(words):
        cnt[idx % n] += 1
        if idx == 0:
            dic[word] = True
            last = word[-1]
            continue

        if last != word[0]:
            answer = [idx % n + 1, cnt[idx % n]]
            return answer

        if word in dic:
            answer = [idx % n + 1, cnt[idx % n]]
            return answer

        else:
            dic[word] = True
            last = word[-1]

    return answer

File: 프로그래머스/2/12981. 영어 끝말잇기/영어 끝말잇기.cpp
#include <string>
#include <vector>
#include <iostream>
#include <unordered_set>

using namespace std;

vector<int> solution(int n, vector<string> words) {
    vector<int> answer(2, 0);
    unordered_set<string> dic;
    string end = "";

    for (int i = 0; i < words.size(); i++) {
        string now = words[i];

        if (i != 0 && end.back() != now.front()) {
            answer = {i % n + 1, (i / n) + 1};
            break;
        }

        if (dic.find(now) != dic.end()) {
            answer = {i % n + 1, (i / n) + 1};
            break;
        }

        dic.insert(now);
        end = now;
    }

    return answer;
}
